### ISSUE TITLE
Add timeout option for transaction creation

### ISSUE THREAD

**Issue by alexquo** [SynapseFI/SynapseNode#84]
We are wondering if there is a timeout option that we can use to timeout transaction creation on your side:
https://docs.synapsefi.com/reference#create-transaction

for example something like this:

```js
    const newTX = await synapseUser.createTransaction(
      achNodeID,
      {
        to: {
          // ...
        },
        amount: {
         // ...
        },
        extra: {
          // ...
        },
        fees: [
          // ...
        ],
      },
      idempotencyKey, {
          timeout: 3000,   // timeout request after 3000 ms.  /// <<<< here
    }
    );
```

is this possible?

**Comment by jenstroeger** [comment]
@alexquo, why would you want to time out, does the response take too long?

**Comment by alexquo** [comment]
yeah at times it appears that this api call takes awhile, and an outside block of our code times out after X seconds, so we would want to timeout your api call after say  X - 10 seconds or something.

**Comment by jenstroeger** [comment]
Ah ok, thank you! I would like to suggest to hold off with this for a couple of months — if possible — because we’re currently addressing these issues on the backend. If you continue to have problems, can you please submit a ticket?
